README.md

```diff
@@ -34,87 +79,69 @@ goe wallet unlock
 goe wallet lock
 ```
 
-**🔑 Note: Other wallet-dependent operations require the wallet to be unlocked.**
-
+> 🔑 **Note:** Wallets are secured using a password-derived key stored in your system keychain.
+> - **Unlock**: Enter your password to derive a key and store it in the system keychain to decrypt your private key for Git operations.
+> - **Lock**: Remove the derived key from the system keychain to secure your wallet.
 
-## 2. Repo CLI
 
-The repo CLI allows you to create repositories, set branches, and manage permissions on-chain.
+## 2. Repo Command
 
-### Commands
+Create and manage on-chain repositories and permissions.
 
+- **Create a repository**
+```bash
+goe repo create <repo_name> --chain-id <chain_id>
+```
 
 - **List repositories**
 ```bash
 goe repo list --chain-id <chain_id>
-
-# eg.
-goe repo list --chain-id  11155111
 ```
 
-
-- **Create a repository**
+- **List branches**
 ```bash
-goe repo create <repo_name> --chain-id <chain_id>
-
-# eg.
-goe repo create test-repo --chain-id 11155111
+goe repo branches <repo_address> --chain-id <chain_id>
 ```
 
 - **Set default branch**
 ```bash
 goe repo default-branch <repo_address> <branch_name> --chain-id <chain_id>
-
-# eg.
-goe repo default-branch 0x0533dc9CD8...aF4279Bda20f55 master --chain-id 11155111
 ```
 
-
-- **Grant push access**
+- **Grant / Revoke push access**
 ```bash
-goe repo grant-push <repo_address> <puser_address> --chain-id <chain_id>
-
-# eg.
-goe repo grant-push 0x0533dc9CD84D...dfaF4279Bda20f55 0x1234... --chain-id 11155111
-```
-
-- **Revoke push access**
-```bash
-goe repo revoke-push <repo_address> <puser_address> --chain-id <chain_id>
-
-# eg.
-goe repo revoke-push 0x0533dc9CD84D...dfaF4279Bda20f55 0x1234... --chain-id 11155111
+goe repo grant-push <repo_address> <user_address> --chain-id <chain_id>
+goe repo revoke-push <repo_address> <user_address> --chain-id <chain_id>
 ```
 
----
-
 
-## 3. Git Helper
+## Example Workflow
 
-The second CLI serves as a Git helper, allowing you to interact with decentralized Git repositories using standard Git commands.
+```bash
+# 1. Create or unlock your wallet
+goe wallet create
+goe wallet unlock
 
-### Protocol
+# 2. Create a new repository on Sepolia
+goe repo create my-project --chain-id 11155111
 
-Repositories are accessed via the **goe://** protocol:
+# 3. Set the default branch
+goe repo default-branch <repo_address> master --chain-id 11155111
```

Comment:
I am not able to go through the command, which reports the follow errors:

```
[9:04:42 AM] Failed to set default branch: execution reverted: "EthfsRepo: Branch not exists"
```

Reply:
Same as above